File: android-exercises/1 - shared_preferences/SharedPreferences_java/app/src/main/java/com/bugrahankaramollaoglu/sharedpreferences_java/MainActivity.java
package com.bugrahankaramollaoglu.sharedpreferences_java;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    TextView textView;
    EditText editText;
    Button saveButton;
    Button deleteButton;
    SharedPreferences sharedPreferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView);
        editText = findViewById(R.id.editText);
        saveButton = findViewById(R.id.saveButton);
        deleteButton = findViewById(R.id.deleteButton);
        
        sharedPreferences = getPreferences(MODE_PRIVATE);
        String nameStored = sharedPreferences.getString("namekey","");
        textView.setText("Your name: " + nameStored);
        
    }

    public void saveData(View view) {

        if (editText.getText().toString().isEmpty()) {
            Toast.makeText(MainActivity.this, "Cannot be empty", Toast.LENGTH_SHORT).show();
        } else {
            String name = editText.getText().toString();
            textView.setText("Your name: " + name);
            
            // save the name to sharedPref
            SharedPreferences.Editor editor = sharedPreferences.edit();
            editor.putString("namekey", name);
            editor.apply();          
            
        }
    }

    public void deleteData(View view) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.remove("namekey");
        editor.apply();
        textView.setText("Your name: ");
    }
}

File: android-exercises/1 - shared_preferences/SharedPreferences_kotlin/app/src/main/java/com/bugrahankaramollaoglu/sharedpreferences_kotlin/MainActivity.kt
package com.bugrahankaramollaoglu.sharedpreferences_kotlin

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.bugrahankaramollaoglu.sharedpreferences_kotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var sharedPreferences: SharedPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        sharedPreferences = this.getSharedPreferences(
            "com.bugrahankaramollaoglu.sharedpreferences_kotlin",
            Context.MODE_PRIVATE
        )

        val nameStored: String = this.sharedPreferences.getString("namekey", "").toString()
        if (nameStored != "") {
            binding.textView.text = "Your name: ${nameStored}"
        } else {
            binding.textView.text = "Your name: "
        }
    }

    fun saveData(view: View) {
        if (binding.editText.text.isEmpty()) {
            Toast.makeText(this@MainActivity, "Cannot be empty", Toast.LENGTH_SHORT).show()
        } else {
            val name: String = binding.editText.text.toString()
            binding.textView.text = name

            // save to sharedPref
            this.sharedPreferences.edit().putString("namekey", name).apply()
            
        }
    }

    fun deleteData(view: View) {
        sharedPreferences.edit().remove("namekey").apply()
        textView.setText("your age: ")
    }
}
